src/main/java/ca/on/oicr/pineryreports/reports/impl/LocationMissingReport.java

```diff
@@ -0,0 +1,177 @@
+
+  public static final String REPORT_NAME = "location-missing";
+  private List<SampleDto> locationMissing = new ArrayList<>();
+
+  private static final Option OPT_AFTER = CommonOptions.after(false);
+  private static final Option OPT_BEFORE = CommonOptions.before(false);
+  private static final Option OPT_PROJECT = CommonOptions.project(true);
+  private static final Option OPT_USER_IDS = CommonOptions.users(false);
+
+  private static final String DATE_REGEX = "\\d{4}-\\d{2}-\\d{2}";
+
+  private static final List<ColumnDefinition> COLUMNS = Collections.unmodifiableList(Arrays.asList(
+      new ColumnDefinition("Sample ID"),
+      new ColumnDefinition("Sample Alias"),
+      new ColumnDefinition("Sample Barcode")));
+
+  private String start;
+  private String end;
+  private String project;
+  private final List<Integer> userIds = new ArrayList<>();
+
+  @Override
+  public String getReportName() {
+    return REPORT_NAME;
+  }
+
+  @Override
+  public Collection<Option> getOptions() {
+    return Sets.newHashSet(OPT_AFTER, OPT_BEFORE, OPT_PROJECT, OPT_USER_IDS);
+  }
+
+  @Override
+  public void processOptions(CommandLine cmd) throws ParseException {
+    if (cmd.hasOption(OPT_AFTER.getLongOpt())) {
+      String after = cmd.getOptionValue(OPT_AFTER.getLongOpt());
+      if (!after.matches(DATE_REGEX)) {
+        throw new ParseException("After date must be in format yyyy-mm-dd");
+      }
+      this.start = after;
+    }
+
+    if (cmd.hasOption(OPT_BEFORE.getLongOpt())) {
+      String before = cmd.getOptionValue(OPT_BEFORE.getLongOpt());
+      if (!before.matches(DATE_REGEX)) {
+        throw new ParseException("Before date must be in format yyyy-mm-dd");
+      }
+      this.end = before;
+    }
+
+    if (cmd.hasOption(OPT_PROJECT.getLongOpt())) {
+      this.project = cmd.getOptionValue(OPT_PROJECT.getLongOpt());
+    }
+
+    if (cmd.hasOption(OPT_USER_IDS.getLongOpt())) {
+      String[] users = cmd.getOptionValue(OPT_USER_IDS.getLongOpt()).split(",");
+      for (String user : users) {
+
+        if (user != null && !"".equals(user)) {
+          userIds.add(Integer.valueOf(user));
+        }
+      }
+    }
+  }
+
+  @Override
+  public String getTitle() {
+    return "Samples with no location for " + (project == null ? "all projects" : project) + " Report";
+  }
+
+  @Override
+  protected void collectData(PineryClient pinery) throws HttpResponseException {
+    List<SampleDto> samples;
+    if (project == null) {
+      samples = pinery.getSample().all();
+    } else {
+      samples = pinery.getSample().allFiltered(new SamplesFilter().withProjects(Lists.newArrayList(project)));
+    }
+    Map<String, SampleDto> allSamples = mapSamplesById(samples);
+
+    locationMissing = filter(samples, Arrays.asList(byNullLocation(), byCreator(userIds), byReceivedBetween(start, end),
+        byNotIdentity(), byReceivedOrChildOfReceived(allSamples), byNonZeroVolume()));
+    if (userIds != null && !userIds.isEmpty()) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2589)